### ISSUE TITLE
false positive for property

### ISSUE THREAD

**Issue by d3matt** [PyCQA/pylint#1510]
### Steps to reproduce

    class Runner:
        '''callable class that might take work to construct'''
        def __call__(self, *args, **kwargs):
            print('called with {} {}'.format(args, kwargs))

    class Runstuff:
        def __init__(self):
            self._runner = None

        def dostuff(self):
            '''call the runner class'''
            self.runner('hello world')

        @property
        def runner(self):
            '''constructs Runner and returns it'''
            if self._runner is None:
                self._runner = Runner()
            return self._runner

### Current behavior
E: 13, 8: self.runner is not callable (not-callable)

### Expected behavior
No error

### pylint --version output
pylint 1.7.1, 
astroid 1.5.2
Python 3.5.3 (default, Apr 24 2017, 13:32:13) 
[GCC 6.3.1 20161221 (Red Hat 6.3.1-1)]

**Comment by mbyrnepr2** [comment]
I don't observe the `not-callable` on:
```python
pylint 2.14.0-dev0
astroid 2.11.4
Python 3.9.10 (v3.9.10:f2f3f53782, Jan 13 2022, 17:02:14) 
[Clang 6.0 (clang-600.0.57)]
```
